test/e2e/resource_test.go

```diff
@@ -183,6 +185,75 @@ var _ = ginkgo.Describe("[resource-status collection] resource status collection
 		})
 	})
 
+	ginkgo.Context("ServiceStatus collection testing", func() {
+		var serviceNamespace, serviceName string
+		var service *corev1.Service
+
+		ginkgo.BeforeEach(func() {
+			policyNamespace = testNamespace
+			policyName = deploymentNamePrefix + rand.String(RandomStrLength)
+			serviceNamespace = testNamespace
+			serviceName = policyName
+
+			service = helper.NewService(serviceNamespace, serviceName)
+			service.Spec.Type = corev1.ServiceTypeNodePort
+			policy = helper.NewPropagationPolicy(policyNamespace, policyName, []policyv1alpha1.ResourceSelector{
+				{
+					APIVersion: service.APIVersion,
+					Kind:       service.Kind,
+					Name:       service.Name,
+				},
+			}, policyv1alpha1.Placement{
+				ClusterAffinity: &policyv1alpha1.ClusterAffinity{
+					ClusterNames: framework.ClusterNames(),
+				},
+			})
+		})
+
+		ginkgo.BeforeEach(func() {
+			framework.CreateService(kubeClient, service)
+			ginkgo.DeferCleanup(func() {
+				framework.RemoveService(kubeClient, serviceNamespace, serviceName)
+			})
+		})
+
+		ginkgo.It("NodePort service apply status collection testing", func() {
+			nodePorts := sets.NewInt32()
+			// simulate the update of the service status in member clusters.
+			ginkgo.By("Update service status in member clusters", func() {
+
+				for _, clusterName := range framework.ClusterNames() {
+					clusterClient := framework.GetClusterClient(clusterName)
+					gomega.Expect(clusterClient).ShouldNot(gomega.BeNil())
+					gomega.Eventually(func(g gomega.Gomega) {
+						memberSvc, err := clusterClient.CoreV1().Services(serviceNamespace).Get(context.TODO(), serviceName, metav1.GetOptions{})
+						g.Expect(err).NotTo(gomega.HaveOccurred())
+						for _, servicePort := range memberSvc.Spec.Ports {
+							nodePorts.Insert(servicePort.NodePort)
+						}
+					}, pollTimeout, pollInterval).Should(gomega.Succeed())
+				}
+				// check service nodePort
+				gomega.Expect(nodePorts.Len() == 1).Should(gomega.BeTrue())
```

Comment:
Sorry, I don't understand why the length of `nodePorts` is equal to 1?

Reply:
I think nodeport should be the same in all member clusters,so, the length of nodePorts is equal to 1

Reply:
The `nodePort` of service is assigned by karmada control-plane, so it seems that we need not to check the service nodePort.

Reply:
> The nodePort of service is assigned by karmada control-plane

yes ,i know, hahaha . double check. 😊